beta/src/components/MDX/Challenges/Challenges.tsx

```diff
@@ -76,32 +73,26 @@ export function Challenges({
   titleId = isRecipes ? 'examples' : 'challenges',
 }: ChallengesProps) {
   const challenges = parseChallengeContents(children);
+  const totalChallenges = challenges.length;
   const scrollAnchorRef = React.useRef<HTMLDivElement>(null);
 
-  const [showHint, setShowHint] = React.useState(false);
-  const [showSolution, setShowSolution] = React.useState(false);
   const [activeChallenge, setActiveChallenge] = React.useState(
     challenges[0].id
   );
 
-  const handleChallengeChange = (challengeId: string) => {
-    setShowHint(false);
-    setShowSolution(false);
-    setActiveChallenge(challengeId);
-  };
-
-  const toggleHint = () => {
-    if (showSolution && !showHint) {
-      setShowSolution(false);
+  React.useEffect(() => {
+    if (scrollAnchorRef.current) {
+      scrollAnchorRef.current.scrollIntoView({
+        block: 'start',
+        behavior: 'smooth',
+      });
     }
-    setShowHint((hint) => !hint);
-  };
+  }, [activeChallenge]);
```

Comment:
The legency logic doesn't work, I'm not sure that it's caused by the `activeChallenge` state leads to re-render, then put these logic in Effect.